jax/_src/pallas/primitives.py

```diff
@@ -493,11 +548,24 @@ def _swap_discharge_rule(in_avals, out_avals, *args_flat, args_tree, **_):
   return (x_new,) + (None,) * (len(in_avals) - 1), out
 
 
-state_discharge.register_discharge_rule(swap_p)(_swap_discharge_rule)
-
-
 def load(x_ref_or_view, idx, *, mask=None, other=None, cache_modifier=None,
          eviction_policy=None, volatile=False) -> jax.Array:
+  """Loads an array from the given index.
+
+  If neither ``mask`` nor ``other`` is specified, this function has the same
+  semantics as ``x[idx]`` in JAX.
+
+  Args:
+    x_ref_or_view: The ref to load from.
+    idx: The indexer to use.
+    mask: An optional boolean mask specifying which indices to load.
+      If mask is ``False`` and ``other`` is not given, no assumptions can
+      be made about the value in the resulting array.
+    other: An optional value to use for indices where mask is ``False``.
+    cache_modifier: TO BE DOCUMENTED.
+    eviction_policy: TO BE DOCUMENTED.
+    volatile: TO BE DOCUMENTED.
```

Comment:
Document the returned value?

Reply:
Actually, it is pretty obvious what that would be, but it is not bad habit to add a line to the docstring about it.

Reply:
Rephrased the docstring to start with "Returns ...".